src/dp_flow.c

```diff
@@ -430,16 +423,16 @@ void dp_process_aged_flows_non_offload(void)
 				DPS_LOG_ERR("Failed to query and remove rte flows", DP_LOG_RET(ret));
 		}
 
-		if (unlikely((current_timestamp - flow_val->timestamp) > timer_hz * flow_val->timeout_value) && (!flow_val->aged))
-			dp_age_out_flow(flow_val);
+		if (unlikely((current_timestamp - flow_val->timestamp) > timer_hz * flow_val->timeout_value))
+			dp_delete_flow_no_flush(next_key, flow_val);
 	}
 }
 
-static __rte_always_inline void dp_remove_flow(struct flow_value *flow_val)
+static __rte_always_inline void dp_remove_flow(struct flow_value *flow_val, const void *key)
 {
 	if (offload_mode_enabled)
 		dp_rte_flow_remove(flow_val);
-	dp_age_out_flow(flow_val);
+	dp_delete_flow_no_flush(key, flow_val);
```

Comment:
I think `no_flush` is only to be used if you do multiple deletions and then you flush manually (that's why `no_flush` is private/static here, flushing needs to happen after deletion otherwise there can be a problem with a cached pointer to the flow structure (that can be freed here).

This can be solved by calling `dp_cntrack_flush_cache()` after the hash-table iteration. But there is high probability that when we change something in one of these hashtable traversals, this will be overlooked and we can get an inconsistent cache state.

I would prefer to just drop the non-flushed version altogether as the only reason for it to exist previously was the code you just removed. I understand that this will be little bit less optimized in terms of performance, but it will be much safer. Especially compared to the previous state where the flushing happened also multiple times per iteration anyway.

Reply:
I am a bit confused what you mean here.
Do you suggest that we remove the "no_flush" from function naming of `dp_delete_flow_no_flush` ? or we delete the function  `dp_delete_flow_no_flush` completely ? in that case, there will be no function to delete the hashtable keys ?

Reply:
I meant move the code from the no-flush version inside the flushing version and only use the flushing version